pkg/controller/registry/resolver/resolver_test.go

```diff
@@ -256,6 +262,139 @@ func multiSourceResolveInstallPlan(t *testing.T, resolver DependencyResolver) {
 	}
 }
 
+func namespaceAndChannelAwareResolveInstallPlan(t *testing.T, resolver DependencyResolver) {
+
+	type csvName struct {
+		name     string
+		owned    []string
+		required []string
+	}
+	var table = []struct {
+		description       string
+		namespace         string
+		mainCSV           string
+		csvs              []csvName
+		crds              []string
+		packageManifests  []registry.PackageManifest
+		existingCSVNames  map[string]struct{}
+		expectedErr       error
+		expectedResources map[registry.ResourceKey]struct{}
+	}{
+		{
+			"MultipleCRDOwnersWithOneExisting",
+			"default",
+			"macaroni-stable",
+			[]csvName{
+				{"macaroni-stable", []string{"macaroni"}, []string{"cheese"}},
+				{"cheese-alpha", []string{"cheese"}, nil},
+				{"cheese-beta", []string{"cheese"}, nil},
+				{"cheese-stable", []string{"cheese"}, nil},
+			},
+			[]string{"macaroni", "cheese"},
+			[]registry.PackageManifest{
+				registry.PackageManifest{
+					PackageName: "cheese",
+					Channels: []registry.PackageChannel{
+						registry.PackageChannel{
+							Name:           "alpha",
+							CurrentCSVName: "cheese-alpha",
+						},
+						registry.PackageChannel{
+							Name:           "beta",
+							CurrentCSVName: "cheese-beta",
+						},
+						registry.PackageChannel{
+							Name:           "stable",
+							CurrentCSVName: "cheese-stable",
+						},
+					},
+					DefaultChannelName: "stable",
+				},
+				registry.PackageManifest{
+					PackageName: "macaroni",
+					Channels: []registry.PackageChannel{
+						registry.PackageChannel{
+							Name:           "stable",
+							CurrentCSVName: "macaroni-stable",
+						},
+					},
+					DefaultChannelName: "stable",
+				},
+			},
+			map[string]struct{}{
+				"cheese-alpha": struct{}{},
+			},
+			nil,
+			map[registry.ResourceKey]struct{}{
+				registry.ResourceKey{Name: "macaroni-stable", Kind: csvKind}: struct{}{},
+				registry.ResourceKey{Name: "macaroni", Kind: crdKind}:        struct{}{},
+				registry.ResourceKey{Name: "cheese-alpha", Kind: csvKind}:    struct{}{},
+				registry.ResourceKey{Name: "cheese", Kind: crdKind}:          struct{}{},
+			},
+		},
+	}
+
+	for _, tt := range table {
```

Comment:
want some test cases for when there are multiple CSVs that own the CRD in the namespace